### PR TITLE
Add a workflow declarations only evaluation method closes #25

### PR DESCRIPTION
**Pull request by Horneth**
I had an old pull request for this that I closed, sorry for the inconvenience.
@mcovarr already gave it a 👍  , wheel chooses @cjllanwarne as second.
It's required to make the new model of input evaluation work in Cromwell.

### CODE REVIEW

**Comment by mcovarr** [comment]
earlier :+1: still in effect

File: src/main/scala/wdl4s/WdlNamespace.scala [hunk 1]
```diff
@@ -122,12 +122,25 @@ case class NamespaceWithWorkflow(importedAs: Option[String],
...
-    val declarationsByScope = workflow.calls.map(_.scopedDeclarations) ++ Seq(workflow.scopedDeclarations)
-    val attemptedEvaluations = declarationsByScope.flatMap(d => d.foldLeft(Map.empty[String, Try[WdlValue]])(evalDeclaration)).toMap
+    val attemptedEvaluations = scopedDeclarations.flatMap(d => d.foldLeft(Map.empty[String, Try[WdlValue]])(evalDeclaration)).toMap
     TryUtil.sequenceMap(attemptedEvaluations)
   }
 
   /**
+    * Evaluates workflow (and only workflow) statically defined declarations sequentially and in order.
+    */
+  def staticWorkflowDeclarationsRecursive(userInputs: WorkflowCoercedInputs, wdlFunctions: WdlStandardLibraryFunctions): Try[WorkflowCoercedInputs] = {
```
Comment (cjllanwarne) [inline src/main/scala/wdl4s/WdlNamespace.scala]: What's `Recursive` about this?

**Comment by cjllanwarne** [comment]
Apart from a naming question, looks good 👍

File: src/main/scala/wdl4s/WdlNamespace.scala (hunk 1 above)
Comment (Horneth) [inline src/main/scala/wdl4s/WdlNamespace.scala]: Nothing, it's a fold actually, I just copied the existing name

**Pull request merged**